RuminantFarmSystems/RuFaS
# Manure application debug

This PR will fix an issue that causes division by zero error in `manure_application.py`.
### Context

Issue(s) closed by this pull request: closes #1715 

      
How big are the changes in the PR? Would you nominate it for a major version upgrade?

### What

A conditional statement has been added to handle the scenario where `new_dry_matter_mass` is zero. In this case, the function will return zero for the new mass, moisture factor, and coverage, as no manure is applied. The test has also been updated to reflect these changes.

### Why

When the initial manure application occurs, there is no manure on the surface. If the surface remainder fraction for this first application is also zero, then no manure from this application will remain on the soil surface. Consequently, the `new_dry_matter_mass` of manure on the surface will stay at 0. This situation is entirely valid and should not trigger a division by zero error.

### How

Adding a condition statement to account for the situation.

### Test plan


1. Switch to the pilot testing branch.
2. Run the simulation with c1f1_wcrops_metadata.json.
3. Errors should no longer appear in the terminal.
